mivideo/mivideo_extract.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import logging
+import os
+import time
+from typing import Union, Sequence, Dict
+
+import pandas as pd
+from google.cloud import bigquery
+from google.oauth2 import service_account
+from sqlalchemy.engine import ResultProxy
+
+import mivideo.queries as queries
+from db.db_creator import DBCreator
+from environ import ENV, CONFIG_DIR
+from run_jobs import ValidDataSourceName
+
+logger = logging.getLogger(__name__)
+
+SHAPE_ROWS: int = 0  # Index of row count in DataFrame.shape() array
+
+
+class MiVideoExtract(object):
+    udpKeyFilePath: str
+    credentials: service_account.Credentials
+    udpDb: bigquery.Client
+    appDb: DBCreator
```

Comment:
I'm curious -- more for my own edification than any particular critique -- why are you using class variables over instance variables here?

Reply:
AFAIK, this is for typing purposes.  They aren't assigned values at this point, so they are not class variables.

That is, that was my intention.  Did I get the syntax wrong?

Reply:
Oh, that makes more sense to me now. I guess I've just been doing my type hinting when defining the instance variable, like here for `Job`: https://github.com/tl-its-umich-edu/course-inventory/blob/29d3ff0df96b95cb4b785302330289545aaab176/run_jobs.py#L26-L34

Reply:
I can change it to that style, if that's preferred.

Since Python typing is embraced more often, I like the style I used because it reminds me of other languages that *require* declarations at that level.  It makes a nice reference to the properties that will be used in the class.

Reply:
I think it's okay to keep it the way you have it, if you prefer it that way. Only drawback is that newbies like me might confuse them for class variables.

Reply:
After more reading, I've decided to move those type declarations out of the class and into `__init__()`.

The way I defined them does make them available as class variables, but since the rest of the code was always accessing them as instance variables, if the class variables had been assigned any values, they would have been ignored.  So, no big deal, really.  It was working the way I intended to.

However, this means that two allocations were made for each variable, in each of the class and instance scopes.  So it's a waste of memory, processing, etc.

Reply:
✅Fixed.